src/main/scala/sigmastate/lang/SigmaTyper.scala

```diff
@@ -361,6 +361,8 @@ class SigmaTyper(val builder: SigmaBuilder) {
       if (i >= 0 && i <= Byte.MaxValue) ByteConstant(i.toByte)
       else error(s"Value $i of type Long cannot be converted to Byte.")
     case v: ContextVariable[_] => v
+    case v: GetVar[_] => v
```

Comment:
@greenhat we should check that id is of type Byte

Reply:
@ergomorphic we do it in http://github.com/ScorexFoundation/sigmastate-interpreter/blob/f04e99518095b7787a6dc97e9a89b9500cb04eef/src/main/scala/sigmastate/lang/SigmaBinder.scala#L102 
I've added a few checks and a throw.